client/src/App.js

```diff
@@ -1,27 +1,55 @@
-import React from 'react'
-import {Route, Switch} from 'react-router-dom'
+import React, { Component } from 'react'
+import { connect } from 'react-redux'
+import { Route, Switch } from 'react-router-dom'
+import PropTypes from 'prop-types'
 
+import { APP_LOAD } from './actions/types'
 import Header from './components/Header/Header'
 import LoginForm from './components/LoginForm/LoginForm'
 import SignupForm from './components/SignupForm/SignupForm'
 import ForgotPasswordForm from './components/ForgotPasswordForm/ForgotPasswordForm'
 import Home from './components/Home/Home'
 import Profile from './components/Profile/Profile'
 import restricted from './components/Restricted/Restricted'
+import api from './api'
 
-function App () {
-  return (
-    <div>
-      <Header />
-      <Switch>
-        <Route path='/login' component={LoginForm}/>
-        <Route path='/signup' component={SignupForm}/>
-        <Route path='/profile' component={restricted(Profile)}/>
-        <Route path='/forgot-password' component={ForgotPasswordForm}/>
-        <Route path='/' component={Home}/>
-      </Switch>
-    </div>
-  )
+class App extends Component {
+  componentWillMount () {
+    // Upon app load, load token from storage and save for use in api
+    const token = window.localStorage.getItem('jwt')
+    if (token) {
+      api.setToken(token)
+    }
+
+    // Load current user from token, if it was loaded
+    this.props.onLoad(token ? api.user.current() : null)
+  }
```

Comment:
Instead of comments, try to extract into a function with a readable name instead. For example,
```js
componentWillMount () {
  this.loadExistingSessionAndUser()
}

loadExistingSessionAndUser () {
  const token = window.localStorage.getItem('jwt')
  if (token) {
    api.setToken(token)
  }
  this.props.onLoad(token ? api.user.current() : null)
}
```